src/main/java/com/google/sps/servlets/EditEventServlet.java

```diff
@@ -0,0 +1,304 @@
+    String day = splitString[2];
+
+    return createDateTime(year, month, day, time, timeZone);
+  }
+
+  private Date parseInputDate(String inputDate, JSONObject json) {
+    String[] splitString = inputDate.split("-");
+    String year = splitString[0];
+    String month = splitString[1];
+    String day = splitString[2];
+
+    return createDate(year, month, day);
+  }
+
+/** Turns inputted strings into a date.*/
+  private Date createDate(String year, String month, String day) {
+    String dateString = month + "-" + day + "-" + year;
+    SimpleDateFormat formatter = new SimpleDateFormat("MM-dd-yyyy");
+    Date dateTime = new Date();
+    try {
+      dateTime = formatter.parse(dateString);
+    } catch (ParseException e) {e.printStackTrace();}
+    return dateTime;
+  }
+
+/** Turns inputted strings into date time. */
+  private Date createDateTime(String year, String month, String day, 
+            String time, String timeZone) {
+    String dateString = month + "-" + day + "-" + year + " " + time + " " + timeZone;
+    SimpleDateFormat formatter = new SimpleDateFormat("MM-dd-yyyy HH:mm z");
+    formatter.setTimeZone(TimeZone.getTimeZone(timeZone));
+    Date dateTime = new Date();
+    try {
+      dateTime = formatter.parse(dateString);
+    } catch (ParseException e) {e.printStackTrace();}
+    return dateTime;
+  }
+
+/** Creates an event entity in datastore. */
+  private void editEntity(
+      HttpServletRequest request, 
+      Date endDateTime, 
+      Date startDateTime, 
+      Date startDate, 
+      Date endDate, 
+      JSONObject json, 
+      String startTime, 
+      String endTime, 
+      String originalStartDate, 
+      String originalEndDate, 
+      String timeZone) {
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String eventName = request.getParameter(Constants.EVENT_NAME_PARAM);
+    String location = request.getParameter(Constants.LOCATION_PARAM);
+    String placeId = request.getParameter(Constants.PLACE_ID_PARAM);
+    String eventDetails = request.getParameter(Constants.EVENT_DETAILS_PARAM);
+    String yesCOVIDSafe = request.getParameter(Constants.COVID_SAFE_PARAM);
+    String privacy = request.getParameter(Constants.PRIVACY_PARAM);
+    String invitedAttendeesString = request.getParameter(Constants.INVITED_ATTENDEES_PARAM);
+    List<String> invitedAttendeesList = Arrays.asList(invitedAttendeesString.split("\\s*,\\s*"));
+    ArrayList<String> invitedAttendees = new ArrayList<String>(invitedAttendeesList);
+    if (invitedAttendeesList.isEmpty()) {
+      invitedAttendees.add(""); // Placeholder entry to prevent empty list from becoming null.
+    }
+    long eventId = Long.parseLong(request.getParameter(Constants.EVENT_ID_PARAM));
+    Query query = new Query(Constants.EVENT_ENTITY_PARAM);
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity eventEntity : results.asIterable()) {
+      if (eventEntity.getKey().getId() == eventId) {
```

Comment:
Instead of iterating over every event in the datastore, what about using a filter to fetch only the event you're interested in?

https://cloud.google.com/appengine/docs/standard/java/datastore/retrieving-query-results#retrieving_entities